common/src/io.cpp

```diff
@@ -274,10 +274,10 @@ pcl::concatenatePointCloud (const pcl::PCLPointCloud2 &cloud1,
       fields2.push_back (field);
     }
 
-    cloud_out.data.resize (nrpts + (cloud2.width * cloud2.height) * cloud_out.point_step);
+    cloud_out.data.resize (nrpts + static_cast<uindex_t>((cloud2.width * cloud2.height)) * cloud_out.point_step);
```

Comment:
No need for the double parenthesis. Did this actually trigger a warning? Usually it only happens on signed to unsigned comparisons. 
It is likely that cloud_out.point_step will also become `index_t` in the future, so you might want to wrap it in this cast as well.

Reply:
Yeah I think only the line underneath threw the error here, and not this one, so I'll remove this cast. As there's no comparison, it shouldn't throw an error. `point_step` is actually `index_t` here, that was one of the changes for this PR.